src/main/java/com/openshift/internal/restclient/model/KubernetesResource.java

```diff
@@ -105,11 +105,11 @@ public void refresh(){
 		this.node = ModelNode.fromJSONString(client.get(getKind(), getName(), getNamespace()).toString());
 	}
 	
-	// TODO Pretty certain this should be protected
 	@Override
 	public ResourceKind getKind(){
-		if(node.has("kind")){
-			return ResourceKind.valueOf(node.get("kind").asString());
+		ModelNode kindNode = get(ResourcePropertyKeys.KIND);
+		if(kindNode.isDefined()){
+			return ResourceKind.valueOf(kindNode.asString());
```

Comment:
@adietish this was one of my thoughts to  KubernetesResource not being abstract for the cases of when we do have kinds we dont understand about and we decide to move away from enum